pkg/apk/apk/implementation.go

```diff
@@ -850,7 +850,10 @@ func DiscoverKeys(ctx context.Context, client *http.Client, auth auth.Authentica
 	if err := auth.AddAuth(ctx, discoveryRequest); err != nil {
 		return nil, err
 	}
-	discoveryResponse, err := client.Do(discoveryRequest)
+
+	rc := retryablehttp.NewClient()
+	rc.HTTPClient = client
+	discoveryResponse, err := rc.StandardClient().Do(discoveryRequest)
```

Comment:
Given the way we wire in APK caching, I am terrified of changes here.

cc @jonjohnsonjr 👀

Reply:
I agree!

Reply:
same but I don't have any specific objections just bad vibes